pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/NonPersistentTopicStatsImpl.java

```diff
@@ -154,8 +155,8 @@ public NonPersistentTopicStatsImpl add(NonPersistentTopicStats ts) {
         Objects.requireNonNull(stats);
         super.add(stats);
         this.msgDropRate += stats.msgDropRate;
-
-        stats.getNonPersistentPublishers().forEach(s -> {
+        IntStream.range(0, stats.getNonPersistentPublishers().size()).forEach(index -> {
```

Comment:
if we care about memory allocations and CPU cycles, I suggest to use a simple for `(int index=0; index < ...)` loop.

Reply:
updated.